js/packages/cli/src/candy-machine-v2-cli.ts

```diff
@@ -1103,109 +1103,124 @@ programCommand('update_existing_nfts_from_latest_cache_file')
     );
   });
 
-programCommand('get_all_mint_addresses').action(async (directory, cmd) => {
-  const { env, cacheName, keypair } = cmd.opts();
+programCommand('get_all_mint_addresses')
+  .option(
+    '-r, --rpc-url <string>',
+    'custom rpc url as the public rpc pool\'s default timeout often won\'t support this operation',
+  )
+  .action(async (directory, cmd) => {
+    const { env, cacheName, keypair, rpcUrl } = cmd.opts();
 
-  const cacheContent = loadCache(cacheName, env);
-  const walletKeyPair = loadWalletKey(keypair);
-  const anchorProgram = await loadCandyProgramV2(walletKeyPair, env);
-
-  const candyMachineId = new PublicKey(cacheContent.program.candyMachine);
-  const [candyMachineAddr] = await deriveCandyMachineV2ProgramAddress(
-    candyMachineId,
-  );
-
-  log.info('Getting mint addresses...');
-  const addresses = await getAddressesByCreatorAddress(
-    candyMachineAddr.toBase58(),
-    anchorProgram.provider.connection,
-  );
-  fs.writeFileSync('./mint-addresses.json', JSON.stringify(addresses, null, 2));
-  log.info('Successfully saved mint addresses to mint-addresses.json');
-});
+    const cacheContent = loadCache(cacheName, env);
+    const walletKeyPair = loadWalletKey(keypair);
+    const anchorProgram = await loadCandyProgramV2(walletKeyPair, env, rpcUrl);
 
-programCommand('get_all_owners_addresses').action(async (directory, cmd) => {
-  const { env, cacheName, keypair } = cmd.opts();
+    const candyMachineId = new PublicKey(cacheContent.program.candyMachine);
+    const [candyMachineAddr] = await deriveCandyMachineV2ProgramAddress(
+      candyMachineId,
+    );
 
-  const cacheContent = loadCache(cacheName, env);
-  const walletKeyPair = loadWalletKey(keypair);
-  const anchorProgram = await loadCandyProgramV2(walletKeyPair, env);
-
-  const candyMachineId = new PublicKey(cacheContent.program.candyMachine);
-  const [candyMachineAddr] = await deriveCandyMachineV2ProgramAddress(
-    candyMachineId,
-  );
-
-  log.info('Getting mint addresses...');
-  const addresses = await getAddressesByCreatorAddress(
-    candyMachineAddr.toBase58(),
-    anchorProgram.provider.connection,
-  );
-
-  log.info('Getting owner addresses...');
-  const owners = await getOwnersByMintAddresses(
-    addresses,
-    anchorProgram.provider.connection,
-  );
-  fs.writeFileSync('./owner-addresses.json', JSON.stringify(owners, null, 2));
-  log.info('Successfully saved owner addresses to owner-addresses.json');
-});
+    log.info('Getting mint addresses...');
+    const addresses = await getAddressesByCreatorAddress(
+      candyMachineAddr.toBase58(),
+      anchorProgram.provider.connection,
+    );
+    fs.writeFileSync('./mint-addresses.json', JSON.stringify(addresses, null, 2));
```

Comment:
I think it would be good to put the CM id or the `c` option value in the file name for those who might reuse the same folder.